packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts

```diff
@@ -80,14 +80,15 @@ export class EcsDeployAction extends Action {
   protected bound(_scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
   codepipeline.ActionConfig {
     // permissions based on CodePipeline documentation:
-    // https://docs.aws.amazon.com/codepipeline/latest/userguide/how-to-custom-role.html#how-to-update-role-new-services
-    options.role.addToPolicy(new iam.PolicyStatement({
+    // https://docs.aws.amazon.com/codepipeline/latest/userguide/security-iam.html#how-to-custom-role
+    options.role.addToPrincipalPolicy(new iam.PolicyStatement({
```

Comment:
Question: what's the difference between `addToPolicy` and `addToPrincipalPolicy` and why is it needed if we're just filling in the missing action?

Reply:
https://github.com/aws/aws-cdk/blob/ba41996939a4c0b06e954c7de7232adef4ce78f5/packages/aws-cdk-lib/aws-iam/lib/principals.ts#L59-L73

`addToPolicy` is deprecated and should use `addToPrincipalPolicy` instead. But I agree we should leave it as is for consistency. I will change it back to `addToPolicy`.